tests/Integration/ClientMiddlewareTest.php

```diff
@@ -140,7 +140,7 @@ public function testThrowOnBrokenConnection() : void
         $client->ping();
 
         $this->expectException(CommunicationFailed::class);
-        $this->expectExceptionMessage('Unable to write request');
+        $this->expectExceptionMessage('Error when writing a request');
```

Comment:
Is it possible to somehow check that the new code works and appends error details to the main message?

Reply:
Done, I've expanded affected error messages in tests to make sure that error information is appended